src/functional/iterator/embedded.py

```diff
@@ -798,13 +794,14 @@ def get_ordered_indices(
     axes: Iterable[Axis], pos: Mapping[Tag, FieldIndex | SparsePositionEntry]
 ) -> tuple[FieldIndex, ...]:
     res: list[FieldIndex] = []
-    sparse_position_tracker: dict[Tag, int] = {}
+    sparse_position_tracker: dict[Tag | int, int] = {}
```

Comment:
This change looks wrong to me, but not sure if I see everything.

Reply:
The change is driven by how the function is called elsewhere, again I simply traced the usage and made it explicit. If this looks wrong then the usage of this function is probably wrong in some places.